### PR TITLE
Add Qcircuits support

### LINKED ISSUE #63: Add support to new qc endpoint
### What is the expected behavior?

Provide programmatic access to the new endpoint.

### PR DESCRIPTION
**Pull request by diego-plan9**
### Summary

Fixes #63

Add support for Qcircuits via a `QcircuitsManager` that is exposed as an attribute of `qiskit.IBMQ`. As with this round of pull requests related to version 2 (#78), the internal implementation is subjected to receive some tweaks short-term and might contain some workarounds while dealing with the transition between api versions.

### CODE REVIEW

File: qiskit/providers/ibmq/qcircuits/__init__.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# -*- coding: utf-8 -*-
+
```
Comment (jaygambetta) [inline qiskit/providers/ibmq/qcircuits/__init__.py]: can we have the folder circuits not circuits

File: qiskit/providers/ibmq/qcircuits/__init__.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Module for interacting with QCircuits."""
+
+from .manager import QcircuitsManager
```
Comment (jaygambetta) [inline qiskit/providers/ibmq/qcircuits/__init__.py]: This should also be CircuitsManager

File: qiskit/providers/ibmq/qcircuits/__init__.py (hunk 1 above)
Comment (diego-plan9) [reply qiskit/providers/ibmq/qcircuits/__init__.py]: Sure - note this file and class is not really exposed to the user (the real binding of the user interface happens in https://github.com/Qiskit/qiskit-ibmq-provider/pull/79/files#diff-fed302e3beba659413d59e360ed8dca3R49), but I'm good changing it!

File: qiskit/providers/ibmq/api/ibmqconnector.py [hunk 3]
```diff
@@ -385,6 +385,33 @@ def available_backends(self):
 
         return response
 
+    def qcircuit_run(self, name, **kwargs):
```
Comment (jaygambetta) [inline qiskit/providers/ibmq/api/ibmqconnector.py]: I really want to drop the q

Comment (diego-plan9) [reply qiskit/providers/ibmq/api/ibmqconnector.py]: Haha noted :+1:

**Pull request merged**